link.go

```diff
@@ -284,63 +328,45 @@ func (l *link) muxClose(ctx context.Context, err *Error, deferred func(), onRXTr
 		Error:  err,
 	}
 
-Loop:
-	for {
-		select {
-		case <-ctx.Done():
-			return
-		case l.session.tx <- fr:
-			// after sending the detach frame, break the read loop
-			break Loop
-		case fr := <-l.rx:
-			// read from link to avoid blocking session.mux
-			switch fr := fr.(type) {
-			case *frames.PerformDetach:
-				if fr.Closed {
-					l.detachReceived = true
-				}
-			case *frames.PerformTransfer:
-				if onRXTransfer != nil {
-					onRXTransfer(*fr)
-				}
-			}
-		case <-l.session.done:
-			if l.doneErr == nil {
-				l.doneErr = l.session.doneErr
-			}
-			return
+	select {
+	case <-ctx.Done():
+		return
+	case l.session.tx <- fr:
+		// frame sent to our session mux
```

Comment:
I'm not going to miss the decentralized pumping code that had to be everywhere. Or am I? Probably not.

Reply:
It's like a thorn.  You notice it's gone but you're better off without it.